### PR TITLE
sanitized facial landmarks, and test for facial landmark sanitizaion

### PR DESCRIPTION
**Pull request by RUTUPARNk**
# ✅ Pull Request: Landmark Coordinate Sanitization

**Related Issue:** [#1460](https://github.com/serengil/deepface/issues/1460)
---
## what has been done

- **Validation Helper**  
  Added `is_valid_landmark()` helper function to verify that landmark coordinates are within valid image boundaries.

- **Sanitization Logic**  
  Updated `extract_faces()` to sanitize landmark coordinates. Invalid values (negative or out-of-bounds) are automatically set to `None`.

- **Comprehensive Tests**  
  Added `tests/test_landmark_sanitization.py` with extensive unit tests to cover:
  - Negative coordinates
  - Out-of-bounds coordinates
  - Malformed landmarks
  - Edge cases with partially valid landmarks

- **Files Modified**
  - `deepface/modules/detection.py` — validation and sanitization logic  
  - `tests/test_landmark_sanitization.py` — new test suite
---
## ✅ How to Test

Run linting and tests locally:

```bash
make lint && pytest
pytest tests/test_landmark_sanitization.py

---

### CODE REVIEW

File: deepface/modules/detection.py [hunk 1]
```diff
@@ -149,22 +173,37 @@ def extract_faces(
         w = min(width - x - 1, int(current_region.w))
         h = min(height - y - 1, int(current_region.h))
 
+        # landmark vaildation
```
Comment (serengil) [inline deepface/modules/detection.py:176]: typo

File: tests/test_landmark_sanitization.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pytest
+from deepface.modules.detection import extract_faces, DetectedFace, FacialAreaRegion
+
+def sanitize_landmarks(region, width, height):
```
Comment (serengil) [inline tests/test_landmark_sanitization.py:5]: why don't we use is_valid_landmark here?

File: tests/test_landmark_sanitization.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        x=10, y=10, w=50, h=50,
+        left_eye=(-5, 20),  # invalid
+        right_eye=(20, 200),  # invalid
+        nose=(30, 30),  # valid
+        mouth_left=(150, 20),  # invalid
+        mouth_right=(20, -10),  # invalid
+        confidence=0.9
+    )
+    landmarks = sanitize_landmarks(region, 100, 100)
+    print("Sanitized landmarks:", landmarks)
```
Comment (serengil) [inline tests/test_landmark_sanitization.py:38]: please use the logger instead of print

**Comment by serengil** [comment]
Linting failed

************* Module deepface.modules.detection
deepface/modules/detection.py:167:0: C0303: Trailing whitespace (trailing-whitespace)

**Comment by RUTUPARNk** [comment]
> Linting failed
> 
> ************* Module deepface.modules.detection deepface/modules/detection.py:167:0: C0303: Trailing whitespace (trailing-whitespace)

fixed

File: tests/test_landmark_sanitization.py (hunk 2 above)
Comment (RUTUPARNk) [reply tests/test_landmark_sanitization.py]: in last commit, i did closure inside `extract_faces` now to do what you asked, `is_valid_landmark` need to move to the module level so it can be imported in test. 

and in `extract_faces`, call it as: 
` if not is_valid_landmark(value, width, height):
     landmarks[key] = None `
is this solution okay?
